### PR TITLE
message logging fixes for publish test results command

### PR DESCRIPTION
**Pull request by AshwiniChalla**

### CODE REVIEW

File: src/Agent.Worker/TestResults/ResultsCommands.cs [hunk 1]
```diff
@@ -280,6 +292,17 @@ private void LoadPublishTestResultsInputs(Dictionary<string, string> eventProper
...
+
+        private void LogPublishTestResultsFailureWarning(Exception ex)
+        {
+            string message = ex.Message;
+            if (ex.InnerException != null)
+            {
+                message += Environment.NewLine;
+                message += ex.InnerException.Message;
+            }
+            _executionContext.Warning(StringUtil.Loc("FailedToPublishTestResults", message));
```
Comment (acesiddhu) [inline src/Agent.Worker/TestResults/ResultsCommands.cs]: is this in framework?

File: src/Agent.Worker/TestResults/ResultsCommands.cs [hunk 2]
```diff
@@ -112,108 +106,126 @@ private void PublishTestResults(IResultReader resultReader)
...
+                    else
+                    {
+                        _executionContext.Warning(StringUtil.Loc("InvalidResultFiles", resultFile, resultReader));
+                    }
                 }
             }
+            catch (Exception ex)
+            {
+                //Do not fail the task.
+                LogPublishTestResultsFailureWarning(ex);
```
Comment (acesiddhu) [inline src/Agent.Worker/TestResults/ResultsCommands.cs]: do we fail the task is all resultfiles weren't published. is the behaviour same as old agent?

**Comment by acesiddhu** [comment]
signed off with comments

File: src/Agent.Worker/TestResults/ResultsCommands.cs (hunk 2 above)
Comment (AshwiniChalla) [inline src/Agent.Worker/TestResults/ResultsCommands.cs]: We will never fail this task. This is the behavior from old agent.

File: src/Agent.Worker/TestResults/ResultsCommands.cs (hunk 1 above)
Comment (AshwiniChalla) [inline src/Agent.Worker/TestResults/ResultsCommands.cs]: Yes.

**Pull request merged**